src/controllers/User/VerifyOtp.js

```diff
@@ -1,31 +1,31 @@
-const Joi = require("@hapi/joi");
-const validator = require("express-joi-validation").createValidator({
-  passError: true,
-});
+  const Joi = require("@hapi/joi");
+  const validator = require("express-joi-validation").createValidator({
+    passError: true,
+  });
+  const User = require("../../models/User");
 
-const User = require("../../models/User");
-const { exist } = require("@hapi/joi");
+  const schema = Joi.object().keys({
+    otp       : Joi.number().required(),
+    userId    : Joi.string().required(),
+  });
 
-const schema = Joi.object().keys({
-  otp       : Joi.number().required(),
-  userId    : Joi.string().required(),
-});
+  exports.validator = validator.body(schema);
 
-exports.validator = validator.body(schema);
+  exports.handler = async (req, res, nex) => {
+    const userId  = req.body.userId;
+    const otp     = req.body.otp;
+    const fieldsToSelect ="otp";
+    try {  
+      const user = await User.find({"otp" : { $in : [req.body.otp]}});
 
-exports.handler = async (req, res, nex) => {
-  const userId  = req.body.userId;
-  const otp     = req.body.otp;
-  const fieldsToSelect ="otp";
-  try {    
-    const user = await User.findById(userId).select(fieldsToSelect);
-    
-    if (user.otp == otp) {
-      res.success("verified", otp);
-    } else {
-      res.fail("otp not verified");
+      if ((user.length)>0) {
```

Comment:
Here add checking for max attempt and if max attempt is greater than configured value block this user id